x-pack/legacy/plugins/ml/server/models/job_service/new_job/categorization/examples.ts

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { chunk } from 'lodash';
+import { CATEGORY_EXAMPLES_SAMPLE_SIZE } from '../../../../../common/constants/new_job';
+import {
+  Token,
+  CategorizationAnalyzer,
+  CategoryFieldExample,
+} from '../../../../../common/types/categories';
+import { callWithRequestType } from '../../../../../common/types/kibana';
+import { ValidationResults } from './validation_results';
+
+const CHUNK_SIZE = 100;
+
+export function categorizationExamplesProvider(callWithRequest: callWithRequestType) {
+  const validationResults = new ValidationResults();
+
+  async function categorizationExamples(
+    indexPatternTitle: string,
+    query: any,
+    size: number,
+    categorizationFieldName: string,
+    timeField: string | undefined,
+    start: number,
+    end: number,
+    analyzer: CategorizationAnalyzer
+  ): Promise<{ examples: CategoryFieldExample[]; error?: any }> {
+    if (timeField !== undefined) {
+      const range = {
+        range: {
+          [timeField]: {
+            gte: start,
+            lt: end,
+            format: 'epoch_millis',
+          },
+        },
+      };
+      if (query.bool === undefined) {
+        query.bool = {};
+      }
+      if (query.bool.filter === undefined) {
+        query.bool.filter = range;
+      } else {
+        if (Array.isArray(query.bool.filter)) {
+          query.bool.filter.push(range);
+        } else {
+          query.bool.filter.range = range;
+        }
+      }
+    }
+
+    const results = await callWithRequest('search', {
+      index: indexPatternTitle,
+      size,
+      body: {
+        _source: categorizationFieldName,
+        query,
+        sort: ['_doc'],
+      },
+    });
+
+    const tempExamples: string[] = results.hits?.hits?.map(
+      (doc: any) => doc._source[categorizationFieldName]
+    );
+
+    validationResults.createNullValueResult(tempExamples);
+
+    const allExamples = tempExamples.filter(
```

Comment:
Isn't it possible for `tempExamples` to be `undefined` here? 🤔 
Might be good to check for that before the `.filter` call.

Reply:
fixed by updating types in 062cdff86885d2156709b5f528c0a90c8d053358